### PR TITLE
Custom image upload for requestGroups

### PR DESCRIPTION
**Pull request by kevin-wangg**
- Can upload a custom image for requestGroups now

### CODE REVIEW

File: client/src/components/utils/imageCrop.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        image.src = url;
+        return image;
+    });
+
+export const getCroppedImg = async (imageSrc: string, pixelCrop: Area): Promise<string> => {
+    const image: HTMLImageElement = await createImage(imageSrc);
+    const canvas = document.createElement("canvas");
+    const ctx = canvas.getContext("2d");
+    const maxSize = Math.max(image.width, image.height);
+    const safeArea = 2 * ((maxSize / 2) * Math.sqrt(2));
```
Comment (guy-ro) [inline client/src/components/utils/imageCrop.ts:17]: Bit weird, can just do maxSize * Math.sqrt(2) I think. What is here seems to be fine but logic is not really straightforward

File: client/src/components/atoms/ImagePicker.tsx [hunk 2]
```diff
@@ -1,25 +1,179 @@
...
+
+    const getImageDimensions = async (file: string): Promise<Dimensions> => {
+        return new Promise((resolved) => {
+            const i = new Image();
+            i.onload = () => {
+                resolved({ width: i.width, height: i.height });
+            };
+            i.src = file;
+        });
+    };
```
Comment (guy-ro) [inline client/src/components/atoms/ImagePicker.tsx:48]: Nit, add line after fn

File: client/src/components/organisms/RequestGroupForm.tsx [hunk 3]
```diff
@@ -27,13 +30,15 @@ const RequestGroupForm: FunctionComponent<Props> = (props: Props) => {
...
     const [image, setImage] = useState("");
+    const [uploadedImg, setUploadedImg] = useState("");
     const [images, setImages] = useState([""]);
     const [requestTypesMap, setRequestTypesMap] = useState<Map<string, RequestTypeData>>(new Map());
     const [nameError, setNameError] = useState("");
     const [descriptionError, setDescriptionError] = useState("");
     const [imageError, setImageError] = useState("");
     const [requestTypesError, setRequestTypesError] = useState("");
     const [loadingRequestGroup, setLoadingRequestGroup] = useState(props.operation === "edit");
+    const [croppedArea, setCroppedArea] = useState<Area>({ width: 0, height: 0, x: 0, y: 0 });
```
Comment (guy-ro) [inline client/src/components/organisms/RequestGroupForm.tsx:41]: Nit, group up the image related states

**Review by guy-ro — APPROVED** [review]
LGTM!

**Review by Puepis — APPROVED** [review]
Looks good to me! Great work!

**Pull request merged**